src/docfx/lib/git/GitUtility.cs

```diff
@@ -70,6 +73,129 @@ public static unsafe List<GitCommit>[] GetCommits(string repoPath, List<string>
             return result;
         }
 
+        /// <summary>
+        /// Set git identity
+        /// </summary>
+        /// <param name="name">The git user name</param>
+        /// <param name="email">The git user email</param>
+        /// <returns>Task status</returns>
+        public static async Task SetIdentity(string name, string email)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(name));
+            Debug.Assert(!string.IsNullOrEmpty(email));
+
+            await ExecuteNonQuery(".", $"config --global user.name \"{name}\"");
+            await ExecuteNonQuery(".", $"config --global user.email \"{email}\"");
+        }
+
+        /// <summary>
+        /// Init git repository
+        /// </summary>
+        /// <param name="cwd">The current working directory</param>
+        /// <returns>Task status</returns>
+        public static async Task Init(string cwd)
+        {
+            if (!Directory.Exists(cwd))
+            {
+                Directory.CreateDirectory(cwd);
+            }
+
+            await ExecuteNonQuery(cwd, "init");
+            await ExecuteNonQuery(cwd, @"commit -m ""Init Commit"" --allow-empty");
+        }
+
+        /// <summary>
+        /// Clone git repository from remote to local
+        /// </summary>
+        /// <param name="cwd">The current working directory</param>
+        /// <param name="remote">The remote url</param>
+        /// <param name="path">The path to clone</param>
+        /// <returns>Task status</returns>
+        public static Task Clone(string cwd, string remote, string path)
+        {
+            Debug.Assert(PathUtility.FolderPathHasInvalidChars(path));
+
+            return ExecuteNonQuery(cwd, $"clone {remote} {path.Replace("\\", "/", StringComparison.Ordinal)}", TimeSpan.FromMinutes(20));
+        }
+
+        /// <summary>
+        /// Fetch update from remote
+        /// </summary>
+        /// <param name="cwd">The current working directory</param>
+        /// <returns>Task status</returns>
+        public static Task Fetch(string cwd)
+            => ExecuteNonQuery(cwd, "fetch", TimeSpan.FromMinutes(3));
+
+        /// <summary>
+        /// Pull update from remote
+        /// </summary>
+        /// <param name="cwd">The current working directory</param>
+        /// <param name="remote">The remote name, default is origin</param>
+        /// <returns>Task status</returns>
+        public static Task Pull(string cwd, string remote = null)
+            => ExecuteNonQuery(cwd, $"pull {remote ?? string.Empty}", TimeSpan.FromMinutes(3));
```

Comment:
>remote ?? string.Empty [](start = 44, length = 22)

why don't we make default value `string.Empty`

Reply:
it dependens, if the "pull" and "pull original" behaviors are not same, then we need to write $"pull {remote ?? origin}"

---
In reply to: [180615824](https://github.com/dotnet/docfx/pull/2611#discussion_r180615824) [](ancestors = 180615824)